flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/source/split/BigQuerySourceSplitAssigner.java

```diff
@@ -0,0 +1,247 @@
+                .map(
+                        result -> {
+                            if (result.getStatus().equals(QueryResultInfo.Status.FAILED)) {
+                                throw new IllegalStateException(
+                                        "The BigQuery query execution failed with errors: "
+                                                + result.getErrorMessages()
+                                                        .orElse(Lists.newArrayList()));
+                            }
+                            String projectId = result.getDestinationProject().get();
+                            String dataset = result.getDestinationDataset().get();
+                            String table = result.getDestinationTable().get();
+                            LOG.info(
+                                    "After BigQuery query execution, switching connect options"
+                                            + " to read from table {}.{}.{}",
+                                    projectId,
+                                    dataset,
+                                    table);
+                            return this.readOptions
+                                    .getBigQueryConnectOptions()
+                                    .toBuilder()
+                                    .setProjectId(projectId)
+                                    .setDataset(dataset)
+                                    .setTable(table)
+                                    .build();
+                        });
+    }
+
+    private Optional<QueryResultInfo> runQuery(String query) {
+        return BigQueryServicesFactory.instance(this.readOptions.getBigQueryConnectOptions())
+                .queryClient()
+                .runQuery(this.readOptions.getQueryExecutionProject(), query);
+    }
+
+    public void open() {
+        LOG.info("BigQuery source split assigner is opening.");
+        if (!initialized) {
+            BigQueryConnectOptions connectionOptions =
+                    fetchOptionsFromQueryRun().orElse(this.readOptions.getBigQueryConnectOptions());
+            try (BigQueryServices.StorageReadClient client =
+                    BigQueryServicesFactory.instance(connectionOptions).storageRead()) {
+                String parent = String.format("projects/%s", connectionOptions.getProjectId());
+
+                String srcTable =
+                        String.format(
+                                "projects/%s/datasets/%s/tables/%s",
+                                connectionOptions.getProjectId(),
+                                connectionOptions.getDataset(),
+                                connectionOptions.getTable());
+
+                // We specify the columns to be projected by adding them to the selected fields,
+                // and set a simple filter to restrict which rows are transmitted.
+                TableReadOptions.Builder optionsBuilder = TableReadOptions.newBuilder();
+
+                readOptions
```

Comment:
Any reason not to use ReadSessionCreator from bigquery-connector-common? This code seems to have some code duplication from there, and it lacks the support for view reading

Reply:
we discussed this previously, and https://github.com/GoogleCloudDataproc/flink-bigquery-connector/issues/58 was created to address this once we get this changes into the main branch.